src/dmclock_server.h

```diff
@@ -1762,7 +1763,7 @@ namespace crimson {
 
 	while (!this->finishing) {
 	  // predicate for cond.wait()
-	  const auto pred = [this] () -> bool { return this->finishing; };
+	  const auto pred = [this] () -> bool { return this->finishing || sched_ahead_when > TimeZero; };
```

Comment:
nit, might want to put `return this->finishing || sched_ahead_when > TimeZero; };` at a new line for better readability, like
```c++
	  const auto pred = [this] () -> bool {
	    return this->finishing || sched_ahead_when > TimeZero;
	  };
```

Reply:
The function has been reworked based on your feedback.